### PR TITLE
Move image logic from services/ to painting/.

### PR DESCRIPTION
**Pull request by Hixie**
This allows the scheduler library to depend on the services library
and the painting library to depend on the scheduler library without
the services library having to depend on the scheduler library.

While I was at it I also cleaned up some of the binding logic: the
licenses logic can now be overridden (and the test library does so),
and the image cache can now be overridden as well.

### CODE REVIEW

File: packages/flutter/lib/src/foundation/binding.dart [hunk 1]
```diff
@@ -26,19 +26,17 @@ typedef Future<Map<String, String>> ServiceExtensionCallback(Map<String, String>
...
-/// various BindingBase mixins (such as [ServicesBinding]). For example, the
-/// Widgets library in flutter introduces a binding called
-/// [WidgetsFlutterBinding]. The relevant library defines how to create
-/// the binding. It could be implied (for example,
-/// [WidgetsFlutterBinding] is automatically started from [runApp]), or
-/// the application might be required to explicitly call the
-/// constructor.
+/// The top-most layer used to write the application will have a concrete class
+/// that inherits from BindingBase and uses all the various [BindingBase] mixins
+/// (such as [ServicesBinding]). For example, the Widgets library in flutter
```
Comment (yjbanov) [inline packages/flutter/lib/src/foundation/binding.dart]: nit: "Flutter"

File: packages/flutter/lib/src/foundation/binding.dart [hunk 2]
```diff
@@ -26,19 +26,17 @@ typedef Future<Map<String, String>> ServiceExtensionCallback(Map<String, String>
...
-/// concrete class that inherits from BindingBase and uses all the
-/// various BindingBase mixins (such as [ServicesBinding]). For example, the
-/// Widgets library in flutter introduces a binding called
-/// [WidgetsFlutterBinding]. The relevant library defines how to create
-/// the binding. It could be implied (for example,
-/// [WidgetsFlutterBinding] is automatically started from [runApp]), or
-/// the application might be required to explicitly call the
-/// constructor.
+/// The top-most layer used to write the application will have a concrete class
+/// that inherits from BindingBase and uses all the various [BindingBase] mixins
```
Comment (yjbanov) [inline packages/flutter/lib/src/foundation/binding.dart]: nit: `[BindingBase]`

**Review by yjbanov — APPROVED** [review]

**Comment by yjbanov** [comment]
lgtm

File: packages/flutter/lib/src/foundation/binding.dart (hunk 2 above)
Comment (Hixie) [reply packages/flutter/lib/src/foundation/binding.dart]: good catch, done

File: packages/flutter/lib/src/foundation/binding.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/lib/src/foundation/binding.dart]: fixed

**Comment by Hixie** [comment]
Thanks for the review; will land on green.

**Pull request merged**